diff --git a/core/modules/system/system.install b/core/modules/system/system.install
index 4a8683e..b0d5fc0 100644
--- a/core/modules/system/system.install
+++ b/core/modules/system/system.install
@@ -239,7 +239,7 @@ function system_requirements($phase) {
     if (!drupal_verify_install_file($conf_path, FILE_NOT_WRITABLE, 'dir')) {
       $conf_errors[] = t("The directory %file is not protected from modifications and poses a security risk. You must change the directory's permissions to be non-writable.", array('%file' => $conf_path));
     }
-    foreach (array('settings.php', 'settings.local.php', 'services.yml') as $conf_file) {
+    foreach (array('settings.php', 'local.settings.php', 'services.yml') as $conf_file) {
       $full_path = $conf_path . '/' . $conf_file;
       if (file_exists($full_path) && !drupal_verify_install_file($full_path, FILE_EXIST|FILE_READABLE|FILE_NOT_WRITABLE)) {
         $conf_errors[] = t("The file %file is not protected from modifications and poses a security risk. You must change the file's permissions to be non-writable.", array('%file' => $full_path));
diff --git a/sites/default/default.settings.php b/sites/default/default.settings.php
index d5d8704..ea6fb17 100644
--- a/sites/default/default.settings.php
+++ b/sites/default/default.settings.php
@@ -639,13 +639,13 @@
 /**
  * Load local development override configuration, if available.
  *
- * Use settings.local.php to override variables on secondary (staging,
+ * Use local.settings.php to override variables on secondary (staging,
  * development, etc) installations of this site. Typically used to disable
  * caching, JavaScript/CSS compression, re-routing of outgoing emails, and
  * other things that should not happen on development and testing sites.
  *
  * Keep this code block at the end of this file to take full effect.
  */
-# if (file_exists(__DIR__ . '/settings.local.php')) {
-#   include __DIR__ . '/settings.local.php';
+# if (file_exists(__DIR__ . '/local.settings.php')) {
+#   include __DIR__ . '/local.settings.php';
 # }
diff --git a/sites/development.services.yml b/sites/development.services.yml
index cc21211..7637af8 100644
--- a/sites/development.services.yml
+++ b/sites/development.services.yml
@@ -1,7 +1,7 @@
 # Local development services.
 #
 # To activate this feature, follow the instructions at the top of the
-# 'example.settings.local.php' file, which sits next to this file.
+# 'example.local.settings.php' file, which sits next to this file.
 services:
   cache.backend.memory:
     class: Drupal\Core\Cache\MemoryBackendFactory
diff --git a/sites/example.settings.local.php b/sites/example.local.settings.php
similarity index 94%
rename from sites/example.settings.local.php
rename to sites/example.local.settings.php
index 7859fe5..ff4e6f3 100644
--- a/sites/example.settings.local.php
+++ b/sites/example.local.settings.php
@@ -5,10 +5,10 @@
  * Local development override configuration feature.
  *
  * To activate this feature, copy and rename it such that its path plus
- * filename is 'sites/example.com/settings.local.php', where example.com
+ * filename is 'sites/example.com/local.settings.php', where example.com
  * is the name of your site. Then, go to the bottom of
  * 'sites/example.com/settings.php' and uncomment the commented lines that
- * mention 'settings.local.php'.
+ * mention 'local.settings.php'.
  */
 
 /**
