diff --git a/core/modules/views/src/Plugin/views/style/Rss.php b/core/modules/views/src/Plugin/views/style/Rss.php
index f597116..70b83e1 100644
--- a/core/modules/views/src/Plugin/views/style/Rss.php
+++ b/core/modules/views/src/Plugin/views/style/Rss.php
@@ -8,7 +8,9 @@
 namespace Drupal\views\Plugin\views\style;
 
 use Drupal\Component\Utility\SafeMarkup;
+use Drupal\Core\Config\ConfigFactoryInterface;
 use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
  * Default style plugin to render an RSS feed.
@@ -26,12 +28,24 @@
 class Rss extends StylePluginBase {
 
   /**
+   * The config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
    * Does the style plugin for itself support to add fields to it's output.
    *
    * @var bool
    */
   protected $usesRowPlugin = TRUE;
 
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+  }
+
   public function attachTo(array &$build, $display_id, $path, $title) {
     $url_options = array();
     $input = $this->view->getExposedInput();
@@ -42,6 +56,10 @@ public function attachTo(array &$build, $display_id, $path, $title) {
 
     $url = _url($this->view->getUrl(NULL, $path), $url_options);
 
+    if ($this->view->display_handler->getOption('sitename_title')) {
+      $title = $this->configFactory->get('system.site')->get('name');
+    }
+
     // Add the RSS icon to the view.
     $this->view->feedIcons[] = [
       '#theme' => 'feed_icon',
@@ -138,4 +156,16 @@ public function render() {
     return $build;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory')
+    );
+  }
+
 }
diff --git a/core/modules/views/src/Tests/Plugin/DisplayFeedTest.php b/core/modules/views/src/Tests/Plugin/DisplayFeedTest.php
index 97e6e99..8c7c0cd 100644
--- a/core/modules/views/src/Tests/Plugin/DisplayFeedTest.php
+++ b/core/modules/views/src/Tests/Plugin/DisplayFeedTest.php
@@ -66,12 +66,14 @@ public function testFeedOutput() {
     $view->getExecutable()->newDisplay('block', NULL, 'test');
     $display = &$view->getDisplay('feed_1');
     $display['display_options']['displays']['test'] = 'test';
+    $display['display_options']['sitename_title'] = 1;
     $view->save();
     // Test the feed display adds a feed icon to the block display.
     $this->drupalPlaceBlock('views_block:test_display_feed-test');
     $this->drupalGet('<front>');
     $feed_icon = $this->cssSelect('div.view-id-test_display_feed a.feed-icon');
     $this->assertTrue(strpos($feed_icon[0]['href'], 'test-feed-display.xml'), 'The feed icon was found.');
+    $this->assertEqual($feed_icon[0]['title'], 'Subscribe to '. $site_name, 'The feed icon title contains the site name.');
   }
 
 }
